### PR TITLE
feat: add resource page

### PR DESCRIPTION
**Pull request by Lama64**
The `admin/students/new` page can be used to add new students.

### CODE REVIEW

File: web/src/routes/admin/(resources)/students/[id]/+page.ts [hunk 1]
```diff
@@ -1,14 +1,25 @@
-import { getStudent } from '$lib/sdk/fetch-client';
+import { getStudent, type Student } from '$lib/sdk/fetch-client';
 import { error } from '@sveltejs/kit';
 import type { PageLoad } from './$types';
 
 export const load = (async ({ params }) => {
+  if (params.id == 'new') {
+    const student: Student = { id: 'new', firstName: '', lastName: '', tags: [] };
```
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/students/[id]/+page.ts:7]: I'd be fine with inlining this

File: web/src/routes/admin/(resources)/students/[id]/+page.svelte [hunk 2]
```diff
@@ -1,5 +1,86 @@
...
+    <div class="max-w-sm mt-7">
+      <TagsMultipleSelect {tags} bind:selectedTagIds />
+    </div>
+    <div class="max-w-sm mt-5"></div>
+    <Button
+      on:click={async () => {
+        try {
+          data.create ? await create() : await update();
+        } catch {
+          error(404, { message: 'Could not update student' });
```
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/students/[id]/+page.svelte:79]: This should not throw a 404. I'd probably go with a 400 here — admittedly that's very generic though

File: web/src/routes/admin/(resources)/students/[id]/+page.svelte [hunk 3]
```diff
@@ -1,5 +1,86 @@
...
+  async function create() {
+    let studentRequestDto: StudentRequestDto = {
+      firstName: student.firstName,
+      lastName: student.lastName,
+      tagIds: selectedTagIds,
+    };
+    try {
+      await createStudent(studentRequestDto);
+    } catch {
+      error(404, { message: 'Could not create student' });
```
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/students/[id]/+page.svelte:27]: Same here

File: web/src/routes/admin/(resources)/students/[id]/+page.svelte [hunk 4]
```diff
@@ -1,5 +1,86 @@
...
+  async function update() {
+    let studentRequestDto: StudentRequestDto = {
+      firstName: student.firstName,
+      lastName: student.lastName,
+      tagIds: selectedTagIds,
+    };
+    try {
+      await updateStudent(student.id, studentRequestDto);
+    } catch {
+      error(404, { message: 'Could not update student' });
```
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/students/[id]/+page.svelte:40]: And here

File: web/src/routes/admin/(resources)/students/[id]/+page.svelte (hunk 2 above)
Comment (danieldietzler) [reply web/src/routes/admin/(resources)/students/[id]/+page.svelte:79]: Actually, why are you even catching the error here and throwing the exact same error again?

File: web/src/lib/components/ui/tags-multiple-select/tags-multiple-select.svelte [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  let selectedTags: Selected<unknown>[] = [];
+  export let selectedTagIds;
+</script>
+
+<Select.Root
+  multiple
+  selected={selectedTags}
+  onSelectedChange={(s) => {
+    s ? (selectedTagIds = s.map((tag) => tag.value)) : (selectedTagIds = []);
+  }}
```
Comment (danieldietzler) [inline web/src/lib/components/ui/tags-multiple-select/tags-multiple-select.svelte:16]:
```suggestion
  onSelectedChange={(s) => (selectedTagIds = s?.map((tag) => tag.value) || [])}
```

File: web/src/routes/admin/(resources)/students/[id]/+page.ts (hunk 1 above)
Comment (Lama64) [reply web/src/routes/admin/(resources)/students/[id]/+page.ts:7]: Personally not a fan of inlining this.

**Review by SuperMarcomen — APPROVED** [review]

**Pull request merged**